tools/code_analyzer/op_dependency.cpp

```diff
@@ -0,0 +1,396 @@
+#include <deque>
+#include <iostream>
+#include <set>
+#include <unordered_map>
+#include <unordered_set>
+
+#include "llvm/Demangle/Demangle.h"
+#include "llvm/IR/CallSite.h"
+#include "llvm/IR/Function.h"
+#include "llvm/IR/GlobalVariable.h"
+#include "llvm/IR/Intrinsics.h"
+#include "llvm/IR/Module.h"
+#include "llvm/Pass.h"
+#include "llvm/Support/CommandLine.h"
+#include "llvm/Support/raw_ostream.h"
+using namespace llvm;
+
+enum OutputFormatType { Dot, YAML };
+static cl::opt<OutputFormatType> OutputFormat(
+    "format",
+    cl::desc("Output format."),
+    cl::values(clEnumValN(Dot, "dot", "print as dot"),
+               clEnumValN(YAML, "yaml", "print as yaml")));
+
+static cl::opt<bool> TransitiveClosure(
+    "closure",
+    cl::desc("Output transitive closure."),
+    cl::init(true));
+
+static cl::opt<int> Verbose(
+    "v",
+    cl::desc("Verbose level"),
+    cl::Hidden,
+    cl::init(0));
+
+static cl::list<std::string> DebugFilters(
+    "df",
+    cl::desc("Debug filter patterns. Example: -df CPUType,at::native"),
+    cl::ZeroOrMore,
+    cl::CommaSeparated);
+
+namespace {
+
+typedef std::set<std::string> SET;
+typedef std::unordered_map<std::string, std::set<std::string>> GRAPH;
+
+// Referenced the logic in llvm-cxxfilt.cpp.
+std::string demangle(const std::string& mangled) {
+  int status;
+  const char* decorated = mangled.c_str();
+  size_t decoratedLength = strlen(decorated);
```

Comment:
I guess you cribbed this straight from llvm-cxxfilt.cpp but it sure is weird seeing you call `strlen` on the `c_str` of a `std::string`

Reply:
haha, yes, I copied from llvm code, I can update it.